src/protocol/boolean/random_bits_generator.rs

```diff
@@ -54,12 +90,15 @@ where
     /// inner members multiple times, I/O errors while executing MPC protocols,
     /// read from an empty buffer, etc.
     pub async fn generate(&self) -> Result<RandomBitsShare<F, S>, Error> {
```

Comment:
I think that this should retain the atomic counter, but only use it for the backup.

Then this can take an index for use with the primary one.